developer-docs-site/docs/move/move-on-aptos/cryptography.md

```diff
@@ -0,0 +1,211 @@
+| [Ed25519](https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-stdlib/sources/cryptography/ed25519.move)                          | Edwards 25519 | 64                | 32              | No           | DLA, ROM    | Fast          |                   |
+| [MultiEd25519](https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-stdlib/sources/cryptography/multi_ed25519.move)               | Edwards 25519 | $4 + t \cdot 64$  | $n \cdot 32$    | No           | DLA, ROM    | Easy-to-adopt | Large sig. size   |
+| MinPK [BLS](https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-stdlib/sources/cryptography/bls12381.move)                       | BLS12-381     | 96                | 48              | No           | CDH, ROM    | Versatile     | Slow verification |
+| MinSig [BLS](https://github.com/aptos-labs/aptos-core/blob/7d4fb98c6604c67e526a96f55668e7add7aaebf6/aptos-move/move-examples/drand/sources/drand.move#L57) | BLS12-381     | 48                | 96              | No           | CDH, ROM    | Versatile     | Slow verification |
+
+**Note:** The [`aptos_std::bls12381`](https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-stdlib/sources/cryptography/bls12381.move) module for [MinPK BLS](https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-bls-signature-05#name-variants) signatures is rather **versatile**: it supports verification of individual signatures, **multi**-signatures, **aggregate** signatures and **threshold** signatures.
+
+The digital signature modules above can be used to build smart contract-based wallets, secure claiming mechanisms for airdrops, or any digital-signature-based access-control mechanism for dapps.
+
+The right choice of a signature scheme in your dapp could depend on many factors:
+1. **Backwards-compatibility** 
+   - e.g., perhaps the dapp's users only sign using Ed25519
+2. **Ease-of-implementation**
+   - e.g., while there exist $t$-out-of-$n$ threshold protocols for Ed25519, they are complex to implement on the signer side. As a result, MultiEd25519 could be adopted due to how easily signing can be implemented.
+3. **Efficiency** 
+   - e.g., some applications are more sensitive to signature size, while others to PK size
+   - e.g., some applications are more sensitive to signing time, while others to signature verification time
+4. **Security analysis**
+   - e.g., ECDSA's security is proven under strong assumptions such as the Generic Group Model (GGM)
+   - e.g., some signature schemes have malleability issues: a valid signature $\sigma$ can be mauled into a _different_ yet-still-valid signature $\sigma'$ on the same message $m$
+5. **Versatility**
+   - e.g., $t$-out-of-$n$ threshold BLS is very simple to implement 
```

Comment:
```suggestion
The right choice of a signature scheme in your dapp could depend on many factors:
1. **Backwards-compatibility** 
   - If your dapp's user base predominantly uses a particular signing mechanism, it would be prudent to support that mechanism for ease of transition and adoption.
     - Example: If users mainly sign using Ed25519, it becomes a logical choice.
2. **Ease-of-implementation**
   - While theoretically sound, complex protocols may be challenging to implement in practice.
     - Example: Even though $t$-out-of-$n$ threshold protocols for Ed25519 exist, their intricacy on the signer's side might push developers toward MultiEd25519 due to its more straightforward signing implementation.
3. **Efficiency** 
   - Depending on the dapp's requirements, you might prioritize one aspect of efficiency over another.
     - Signature Size vs. Public Key (PK) Size: Some applications might prioritize a smaller signature footprint, while others might emphasize a compact PK size.
     - Signing Time vs. Verification Time: For certain dapps, the signing speed might be more crucial, while for others, rapid signature verification could be the priority.
4. **Security analysis**
   - It is essential to consider the underlying assumptions and potential vulnerabilities of a signature scheme.
     - Example: ECDSA's security is proven under strong assumptions such as the Generic Group Model (GGM).
     - Malleability Concerns: Some signature schemes are susceptible to malleability, where a valid signature, `$\sigma$`, can be mauled into a different yet still valid signature, `$\sigma'$`, for the same message, `$m$`.
5. **Versatility**
   - The adaptability and flexibility of signature schemes are important to consider so you may properly accommodate the cryptographic needs of your dapp.
     - Example: The $t$-out-of-$n$ threshold BLS is very simple to implement.
```

Reply:
I added a short description for each factor and reformatted the examples for improved readability.